src/lib/utilities/is-promise.ts

```diff
@@ -0,0 +1,11 @@
+export function isPromise(promiseTentially: any): Boolean {
```

Comment:
The type should probably be `unknown` as opposed to any. And we probably want to use the `is` operator the the response.

```js
function isPromise(obj: unknown) obj is Promise<unknown> {
  return !!obj && (typeof obj === 'object' || typeof obj === 'function') && typeof obj.then === 'function';
}
```